### PR TITLE
implement seqframe_idx and frame_idx.

### PR DESCRIPTION
**Pull request by csukuangfj**

### CODE REVIEW

File: k2/csrc/fsa_utils.cu [hunk 1]
```diff
@@ -1816,5 +1816,39 @@ void FixNumStates(FsaVec *fsas) {
...
+      context, dim, set_idx, (int32_t i)->void {
+        int32_t idx01 = arc_map_b_data[i] / num_cols;
+        if (accessor(idx01, 0) != 0) {
+          seq_frame_idx_data[i] = idx01;
+          int32_t fsa_idx0 = row_ids1_data[idx01];
+          frame_idx_data[i] = idx01 - row_splits1_data[fsa_idx0];
+          return;
+        }
+
+        // this arc enters the final state
```
Comment (csukuangfj) [inline k2/csrc/fsa_utils.cu]: I am not sure how to handle arcs entering the final state, so -1 is used here.

File: k2/python/tests/intersect_dense_pruned_test.py [hunk 2]
```diff
@@ -197,7 +221,7 @@ def test_two_fsas_long_pruned(self):
                                   device=device,
                                   requires_grad=True)
 
-            supervision_segments = torch.tensor([[0, 0, 95], [1, 20, 50]],
+            supervision_segments = torch.tensor([[0, 1, 95], [1, 20, 50]],
```
Comment (csukuangfj) [inline k2/python/tests/intersect_dense_pruned_test.py:224]: The supervision segment has a start offset. So 0 is relative to the supervision segment, not to nnet_output. Do we need to add the start offset to `frame_idx`?

**Comment by danpovey** [comment]
I think there's a lot of unnecessary C++.  Can't this all be done from Python?

**Comment by csukuangfj** [comment]
I thought we need to handle arcs entering the final state especially and it is more efficient to compute seqframe_idx and frame_idx in one kernel.  Will implement it in Python.

**Comment by danpovey** [comment]
no, better to treat those arcs the same as others.  If we want to treat them specially it can be done later on by testing the label.
Also the user may not request both of those things.

**Comment by csukuangfj** [comment]
Now everything is moved to Python.

**Comment by danpovey** [comment]
Great- thanks!
